Add Remove all posts for user from room endpoint

majestrate opened the pull request:
* Unit tests
* Fix issue with Room.delete_all_posts where exception thrown when no files uploaded

fixes #67

Commit:
Add Remove all posts for user from room endpoint

* Unit tests
* Fix issue with Room.delete_all_posts where exception thrown when no files uploaded

jagerman commented on sogs/routes/rooms.py:
```diff
@@ -1,5 +1,6 @@
 from .. import config, db, http
 from ..model import room as mroom
+from ..model import user as muser
```
```suggestion
from ..model import room as mroom, user as muser
```

jagerman committed:
Don't set room to null for expired images

Expiring them (also changed here to 0, rather that current time) is
enough to make them inaccessible now.
